server/src/protocol-translation.ts

```diff
@@ -18,8 +18,20 @@ export function uriToPath(stringUri: string): string | undefined {
     return uri.fsPath;
 }
 
+const parseURI = (filepath: string): URI => {
```

Comment:
@rchl I think this is better now - no weird regex stuff 😄  `tsserver` will now either emit filepaths (as before) or URIs. We attempt to parse the filepath and if it's a URI we respect that and pass it upstream, otherwise turn it into a URI with a `file://` scheme

Reply:
I will think about it more. Need to check where it's used and how.